mount/mount.go

```diff
@@ -22,3 +47,48 @@ func All(mounts []Mount, target string) error {
 	}
 	return nil
 }
+
+// TempMount mounts the provided mounts to a temp dir, and pass the temp dir to f.
+// The mounts are valid during the call to the f.
+// Finally we will unmount and remove the temp dir regardless of the result of f.
+func TempMount(mounts []Mount, m Mounter, f func(root string) error) error {
```

Comment:
A big +1 for this idea, I think it makes the calling code read better. This is basically the same pattern as [python's context manager](http://book.pythontips.com/en/latest/context_managers.html).

I see the name was already discussed, but `TransientMount` sounds a little better to me. Other potential names (borrowing from python's naming) could be something like `WithTransientMount`, or `WithTempMount`.

Reply:
I like `WithTempMount`, I agree that is a better naming scheme. I don't think transient is a better term then temp in this case though, they have similar meaning and there are cases where transient is a better word (for example when state changing is involve).

Reply:
done.   use WithTempMount